inference-engine/src/inference_engine/include/openvino/runtime/tensor.hpp

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the OpenVINO Runtime tensor API
+ *
+ * @file openvino/runtime/tensor.hpp
+ */
+#pragma once
+
+#include "common.hpp"
+#include "ie_api.h"
+#include "openvino/core/partial_shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+namespace InferenceEngine {
+class Blob;
+}  // namespace InferenceEngine
+
+namespace ov {
+namespace runtime {
+
+class InferRequest;
+class RemoteTensor;
+class RemoteContext;
+
+/**
+ * @brief Memory access and interpretation API
+ *
+ * It can throw exceptions safely for the application, where it is properly handled.
+ */
+class INFERENCE_ENGINE_API_CLASS(Tensor) {
+    std::shared_ptr<void> _so;
+    element::Type _type;
+    PartialShape _shape;
+    std::shared_ptr<ie::Blob> _impl;
+
+    /**
+     * @brief Constructs Tensor from the initialized std::shared_ptr
+     * @param so Plugin to use. This is required to ensure that Tensor can work properly even if plugin object is
+     * destroyed.
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param impl Initialized shared pointer
+     */
+    Tensor(const std::shared_ptr<void>& so,
+           const element::Type type,
+           const PartialShape shape,
+           const std::shared_ptr<ie::Blob>& impl);
+
+    friend class ov::runtime::InferRequest;
+    friend class ov::runtime::RemoteTensor;
+    friend class ov::runtime::RemoteContext;
+
+public:
+    /**
+     * @brief Default constructor
+     */
+    Tensor() = default;
+
+    /**
+     * @brief Constructs Tensor using element type and shape. Allocate internal host storage using default allocator
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     */
+    Tensor(const element::Type type, const PartialShape shape);
+    /**
+     * @brief Constructs Tensor using element type and shape. Wraps allocated host memory.
+     * @note Dose not own the memory
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param ptr Pointer to allocate memory
+     */
+    Tensor(const element::Type type, const PartialShape shape, void* ptr);
+    /**
+     * @return tensor element type
+     */
+    element::Type get_element_type() const;
+
+    /**
+     * @return tensor element type
+     */
+    PartialShape get_shape() const;
+
+    /**
+     * @return host pointer to tensor memory
+     * @param type Optional type parameter.
+     * If type parameter is specified throws exception
+     * if specified type dose not match with tensor element type
+     */
+    void* data(const element::Type type = element::undefined) const;
+
+    /**
+     * @return host pointer to tensor memory casted to specified type.
+     * Throws exception if specified type dose not match with tensor element type
+     */
+    template <typename T>
+    T* as() const {
+        return static_cast<T*>(data(element::from<T>()));
+    }
```

Comment:
I don't see the following methods:
1. `create_roi`
2. `get_strides`
3. `empty` or maybe `operator bool`
4. `size`, `byte_size`, `element_size`

Reply:
implemented except `create_roi`